cli/lsp/documents.rs

```diff
@@ -920,22 +943,40 @@ impl Documents {
     &self,
     specifiers: Vec<String>,
     referrer: &ModuleSpecifier,
+    maybe_npm_resolver: Option<&NpmPackageResolver>,
```

Comment:
Feels like resolver should be stored on the struct instead?

Reply:
It does, but when snapshotting the lsp, the resolver becomes "snapshotted", so we pass that one here instead of the main resolver. There's some overall improvement (big refactor) that could be made WRT documents and snapshotting to make this more clear, but I think this is ok for now. I opened https://github.com/denoland/deno/issues/16369